### ISSUE TITLE
fix: wrong InputShape/OutputShape in 8 neural network test files

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1037]
## Category: Test Configuration — Wrong InputShape/OutputShape

**Priority:** High — blocks all other invariant tests for these models
**Difficulty:** Easy — only edit test files, no model code changes
**Estimated time:** 1-2 hours

### Problem

8 neural network test files use incorrect `InputShape` or `OutputShape`, causing output dimension mismatches or loss function errors. Every invariant test fails for these models because the test creates input/target tensors with wrong dimensions.

### Models to Fix

| Model | Test File | Current Shape | Correct Shape | Error |
|-------|-----------|---------------|---------------|-------|
| **BGE** | `BGETests.cs` | Default `[1,4]`/`[1,1]` | `InputShape => [768]`, `OutputShape => [768]` | Predicted/actual vector length mismatch |
| **ColBERT** | `ColBERTTests.cs` | Default `[1,4]`/`[1,1]` | Needs investigation of actual output dim | Predicted/actual vector length mismatch |
| **TransformerEmbeddingNetwork** | `TransformerEmbeddingNetworkTests.cs` | Default `[1,4]`/`[1,1]` | `OutputShape => [768]` | Output 768 vs expected 1 |
| **CapsuleNetwork** | `CapsuleNetworkTests.cs` | `[1,28,28]`/`[10]` | `OutputShape => [784]` | Output 784 vs expected 10 |
| **VoxelCNN** | `VoxelCNNTests.cs` | `[1,32,32,32]`/`[1]` | `OutputShape => [128]` (match 3D conv output) | Output 128 vs expected 1 |
| **AudioVisualCorrespondenceNetwork** | `AudioVisualCorrespondenceNetworkTests.cs` | `[512]`/`[1]` | `OutputShape => [512]` | Output 512 vs expected 1 |
| **DenseNetNetwork** | `DenseNetNetworkTests.cs` | Default `[1,4]` | `InputShape => [1, 3, 32, 32]` | Conv layer needs 3D+ input |
| **MobileNetV3Network** | `MobileNetV3NetworkTests.cs` | Default `[1,4]` | `InputShape => [1, 3, 224, 224]` | Conv layer needs 3D+ input |

### How to Fix

1. Open the test file in `tests/AiDotNet.Tests/ModelFamilyTests/NeuralNetworks/`
2. Check what the model's actual output shape is by looking at its architecture or running `model.Predict(input).Shape`
3. Update `InputShape` and `OutputShape` overrides to match
4. Run the test to verify: `dotnet test --filter "FullyQualifiedName~ModelFamilyTests.NeuralNetworks.{Model}Tests"`

### Example Fix

```csharp
// BEFORE (wrong)
public class BGETests : NeuralNetworkModelTestBase
{
    protected override INeuralNetworkModel<double> CreateNetwork() => new BGE<double>();
}

// AFTER (correct)
public class BGETests : NeuralNetworkModelTestBase
{
    protected override int[] InputShape => [768];
    protected override int[] OutputShape => [768];
    protected override INeuralNetworkModel<double> CreateNetwork() => new BGE<double>();
}
```

### Verification

After fixing, all these tests should pass:
- `ForwardPass_ShouldProduceFiniteOutput`
- `OutputDimension_ShouldMatchExpectedShape`
- `Training_ShouldReduceLoss`
- `Training_ShouldChangeParameters`
